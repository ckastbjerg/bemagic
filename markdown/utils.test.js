/* eslint-env jest */
const utils = require('./utils');

test('capitalize works', () => {
    const actual = utils.capitalize('some sentence');
    expect(actual).toMatchSnapshot();
});

test('description works', () => {
    const actual = utils.description('some sentence');
    expect(actual).toMatchSnapshot();
});

test('menuItem works', () => {
    const actual = utils.menuItem('some sentence');
    expect(actual).toMatchSnapshot();
});

test('notice works', () => {
    const actual = utils.notice('some sentence');
    expect(actual).toMatchSnapshot();
});

test('sectionTitle works', () => {
    const actual = utils.sectionTitle('some title');
    expect(actual).toMatchSnapshot();
});

test('title works', () => {
    const actual = utils.title({
        name: 'block',
        classes: 'prefix-block',
    });
    expect(actual).toMatchSnapshot();
});

test('it throws if parameter `list` is not an array', () => {
    expect(() => utils.listString({ list: '' })).toThrow();
});

test('it works', () => {
    expect(utils.listString({
        list: ['inverse', 'secondary'],
    })).toMatchSnapshot();
});

test('it works with custom word', () => {
    expect(utils.listString({
        list: ['inverse', 'secondary'],
        word: 'or',
    })).toMatchSnapshot();
});
