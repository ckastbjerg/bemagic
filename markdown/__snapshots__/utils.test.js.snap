exports[`test capitalize works 1`] = `"Some sentence"`;

exports[`test description works 1`] = `"Some sentence."`;

exports[`test it works 1`] = `"\`inverse\` and \`secondary\`."`;

exports[`test it works with custom word 1`] = `"\`inverse\` or \`secondary\`."`;

exports[`test menuItem works 1`] = `"- **[Some sentence](#some-sentence)**"`;

exports[`test notice works 1`] = `undefined`;

exports[`test sectionTitle works 1`] = `
"
        <a name=some title></a>
        #### Some title
    "
`;

exports[`test title works 1`] = `"## Block \`prefix-block\`"`;
